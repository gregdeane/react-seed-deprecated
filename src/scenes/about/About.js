import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router';
import { connect } from 'react-redux';
import Button from 'react-bootstrap/lib/Button';

import Page from 'components/page/Page';
import { loadAbout } from './reducers/about-reducer';

import './about.scss';

export class About extends Component {
  static propTypes = {
    about: PropTypes.object,
    dispatch: PropTypes.func.isRequired,
  };

  static mapStateToProps(state) {
    return {
      about: state.about,
    };
  }

  componentWillMount() {
    const { dispatch } = this.props;
    dispatch(loadAbout());
  }

  render() {
    const { about } = this.props;

    return (
      <Page
        documentTitle="About"
        title="About"
      >
        <section className="about">
          <p>{about.name}</p>
          <p>{about.html_url}</p>
          <Button>Default</Button>
          <Button bsStyle="primary">Primary</Button>
          <Button bsStyle="danger">Danger</Button>
        </section>
      </Page>
    );
  }
}

export default withRouter(connect(
  About.mapStateToProps,
)(About));
