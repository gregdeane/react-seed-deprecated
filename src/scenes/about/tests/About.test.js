import React from 'react';
import { render } from 'enzyme';

import { About } from '../About';

const getWrapper = props => (
  render(
    <About
      {...props}
      dispatch={() => {}}
    />,
  )
);

const data = {
  name: 'name',
  github: 'github',
};

describe('<About />', () => {
  it('should render default component', () => {
    const wrapper = getWrapper({ about: data });
    const element = wrapper.find('.page-title');

    expect(element.text()).toEqual('About');
    expect(wrapper).toMatchSnapshot();
  });

  it('should return state', () => {
    const state = About.mapStateToProps({ about: data });

    expect(state.about).toEqual({ ...data });
  });
});
