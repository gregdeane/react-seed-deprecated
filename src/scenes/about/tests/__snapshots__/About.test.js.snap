// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<About /> should render default component 1`] = `
<main
  class="page"
>
  <h1
    class="page-title"
  >
    About
  </h1>
  <div
    class="page__content"
  >
    <section
      class="about"
    >
      <p>
        name
      </p>
      <p />
      <button
        class="btn btn-default"
        type="button"
      >
        Default
      </button>
      <button
        class="btn btn-primary"
        type="button"
      >
        Primary
      </button>
      <button
        class="btn btn-danger"
        type="button"
      >
        Danger
      </button>
    </section>
  </div>
</main>
`;
